nemo/collections/nlp/parts/nlp_overrides.py

```diff
@@ -434,12 +434,9 @@ class NLPSaveRestoreConnector(SaveRestoreConnector):
     def __init__(self) -> None:
         if not HAVE_APEX:
             logging.warning(
-                "Apex was not found. Please see the NeMo README for installation instructions: https://github.com/NVIDIA/apex\n"
+                "Only support single gpu inference when Apex was not found. Please see the NeMo README for installation instructions: https://github.com/NVIDIA/apex\n"
```

Comment:
The error message seems unnecessary, and incomplete. You first need to warn the user that they haven't installed Apex, then the rest of the message.

Reply:
@ericharper any suggestions? That no apex error/warning message occurs a lot of places, we may need to rephrase that as @titu1994 suggested?